tensorflow/tools/ci_build/linux/mkl/set-build-env.py

```diff
@@ -54,53 +31,173 @@
     "--linkopt=-zrelro", "--linkopt=-znow", "--linkopt=-fstack-protector"
 ]
 
+class IntelPlatform(object):
+  min_gcc_major_version_ = 0
+  min_gcc_minor_version_ = 0
+  host_gcc_major_version_ = 0
+  host_gcc_minor_version_ = 0
+  BAZEL_PREFIX_ = "--copt="
+  ARCH_PREFIX_ = "-march="
+  FLAG_PREFIX_ = "-m"
+
+  def __init__(self, min_gcc_major_version, min_gcc_minor_version):
+    self.min_gcc_minor_version_ = min_gcc_minor_version
+    self.min_gcc_major_version_ = min_gcc_major_version
+
+  # Return True or False depending on whether
+  # The platform optimization flags can be generated by
+  # the gcc version specified in the parameters
+  def set_host_gcc_version(self, gcc_major_version, gcc_minor_version):
+    # True only if the gcc version in the tuple is >=
+    # min_gcc_major_version_, min_gcc_minor_version_
+    if gcc_major_version < self.min_gcc_major_version_:
+      print("Your MAJOR version of GCC is too old: {}; "
+            "it must be at least {}.{}".format(gcc_major_version,
+                                                self.min_gcc_major_version_,
+                                                self.min_gcc_minor_version_))
+      return False
+    elif gcc_major_version == self.min_gcc_major_version_ and \
+          gcc_minor_version < self.min_gcc_minor_version_:
+      print("Your MINOR version of GCC is too old: {}; "
+        "it must be at least {}.{}".format(gcc_minor_version,
+                                            self.min_gcc_major_version_,
+                                            self.min_gcc_minor_version_))
+      return False
+    print("gcc version OK: {}.{}".format(gcc_major_version, gcc_minor_version))
+    self.host_gcc_major_version_ = gcc_major_version
+    self.host_gcc_minor_version_ = gcc_minor_version
+    return True
+
+  # return a string with all the necessary bazel formatted flags for this
+  # platform in this gcc environment
+  def get_bazel_gcc_flags(self):
+    raise NotImplementedError(self)
+
+  # return true or false depending on whether the host gcc version
+  # is newer or older than the gcc version in which the new
```

Comment:
This returns true when the host version is older (not newer). Please update the description.

Reply:
Thank you for keeping me honest and precise!